Granny/Analyses/Segmentation.py

```diff
@@ -172,7 +173,18 @@ def __init__(self):
             )
         )
 
-        self.addInParam(self.model, self.input_images)
+        # Add confidence threshold parameter for YOLO predictions
+        self.confidence_threshold = FloatValue(
+            "conf_threshold",
+            "confidence_threshold", 
+            "Minimum confidence threshold for YOLO detections. Objects with confidence below this value will be filtered out."
+        )
+        self.confidence_threshold.setValue(0.25)  # YOLO default
+        self.confidence_threshold.setMin(0.0)
+        self.confidence_threshold.setMax(1.0)
+        self.confidence_threshold.setIsRequired(False)
+
+        self.addInParam(self.model, self.input_images, self.confidence_threshold)
```

Comment:
Looks good, just test with a high threhsold that you do see some segements falling out.